config/translate.go

```diff
@@ -165,25 +180,53 @@ func TranslateFromV1(old v1.Config) types.Config {
 	return config
 }
 
+// golang--
+func translateV1MkfsOptionsToV2_1OptionSlice(opts v1.MkfsOptions) []types.Option {
+	newOpts := make([]types.Option, len(opts))
+	for i, o := range opts {
+		newOpts[i] = types.Option(o)
+	}
+	return newOpts
+}
+
+// golang--
+func translateStringSliceToV2_1SSHAuthorizedKeySlice(keys []string) []types.SSHAuthorizedKey {
+	newKeys := make([]types.SSHAuthorizedKey, len(keys))
+	for i, k := range keys {
+		newKeys[i] = types.SSHAuthorizedKey(k)
+	}
+	return newKeys
+}
+
+// golang--
+func translateStringSliceToV2_1UsercreateGroupSlice(groups []string) []types.UsercreateGroup {
+	var newGroups []types.UsercreateGroup
```

Comment:
Because a preallocated slice of zero length is different than a declared slice that was never appended to, and the tests break for specifically this field when it compares the two.

The other two are preallocated because the tests didn't complain on those, I like preallocated a lot more than constantly reallocating, and imo it's syntactically just as easy for humans to read.